src/builder/QuarkBuilder.sol

```diff
@@ -49,6 +49,160 @@ contract QuarkBuilder {
 
     /* ===== Main Implementation ===== */
 
+    struct CometSupplyIntent {
+        uint256 amount;
+        string assetSymbol;
+        uint256 blockTimestamp;
+        uint256 chainId;
+        address comet;
+        address sender;
+    }
+
+    function cometSupply(
+        CometSupplyIntent memory cometSupplyIntent,
+        Accounts.ChainAccounts[] memory chainAccountsList,
+        PaymentInfo.Payment memory payment
+    ) external pure returns (BuilderResult memory /* builderResult */ ) {
+        assertSufficientFunds(cometSupplyIntent.assetSymbol, cometSupplyIntent.amount, chainAccountsList);
+        assertFundsAvailable(
+            cometSupplyIntent.chainId,
+            cometSupplyIntent.assetSymbol,
+            cometSupplyIntent.amount,
+            chainAccountsList,
+            payment
+        );
+
+        uint256 actionIndex = 0;
+        Actions.Action[] memory actions = new Actions.Action[](chainAccountsList.length);
+        IQuarkWallet.QuarkOperation[] memory quarkOperations =
+            new IQuarkWallet.QuarkOperation[](chainAccountsList.length);
+
+        if (
```

Comment:
The bridging logic on this function and transfer are nearly identical, but it's somewhat awkward to abstract the functionality out into a separate function since we're altering the `actions` and `quarkOperations` arrays.

It's probably possible to have a helper function that returns two arrays and then iterates over the return values, pushing the returned values onto `actions` and `quarkOperations`; i'll play around with it

Reply:
this will likely be updated to reflect Kevin's changes from: https://github.com/compound-finance/quark-scripts/pull/30/files#diff-9111113812791cbe6884c8fad436a6674788f2e44d1616c9bb124f30a33bd250R85